### PR TITLE
[Swift] Add Swift 5.0.1 release tags

### PR DESCRIPTION
**Pull request by shahmishal**

### CODE REVIEW

File: library/swift [hunk 1]
```diff
@@ -7,11 +7,19 @@ Maintainers: Ted Kremenek <kremenek@apple.com> (@tkremenek),
...
-Tags: 5.0, 5.0-bionic, bionic, latest
+Tags: 5.0.1, 5.0.1-bionic, bionic, latest
+GitCommit: 0aafffef619fb3b1824c968cbbe2fba4ba41bd26
+Directory: 5.0/ubuntu/18.04
+
+Tags: 5.0.1-xenial, xenial
+GitCommit: 0aafffef619fb3b1824c968cbbe2fba4ba41bd26
+Directory: 5.0/ubuntu/16.04
+
+Tags: 5.0, 5.0-bionic
```
Comment (tianon) [inline library/swift]: Any particular reason to keep this version listed here?  Removing it here will remove it from the "Supported" section on the Hub readme (and will prevent us from spending cycles rebuilding it on the official build server), but the tag will still be available to users who want it.  (See https://github.com/docker-library/official-images#library-definition-files for more detail on this.)

Comment (shahmishal) [reply library/swift]: Updated the file by dropping dot releases.

**Comment by tianon** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-arches b/_bashbrew-arches
index 1f61589..75f9d95 100644
--- a/_bashbrew-arches
+++ b/_bashbrew-arches
@@ -2,8 +2,5 @@ swift:3 @ amd64
 swift:4 @ amd64
 swift:4.0 @ amd64
 swift:4.1 @ amd64
-swift:4.2.1 @ amd64
-swift:4.2.2 @ amd64
-swift:4.2.3 @ amd64
 swift:latest @ amd64
 swift:xenial @ amd64
diff --git a/_bashbrew-list b/_bashbrew-list
index 5e7a50b..7080316 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -7,13 +7,11 @@ swift:4.0.3
 swift:4.1
 swift:4.1.3
 swift:4.2
-swift:4.2.1
-swift:4.2.2
-swift:4.2.3
 swift:4.2.4
 swift:5.0
-swift:5.0-bionic
-swift:5.0-xenial
+swift:5.0.1
+swift:5.0.1-bionic
+swift:5.0.1-xenial
 swift:bionic
 swift:latest
 swift:xenial
diff --git a/swift_4.2.1/Dockerfile b/swift_4.2.1/Dockerfile
deleted file mode 100644
index ade1634..0000000
diff --git a/swift_4.2.2/Dockerfile b/swift_4.2.2/Dockerfile
deleted file mode 100644
index 5af541f..0000000
diff --git a/swift_4.2.3/Dockerfile b/swift_4.2.3/Dockerfile
deleted file mode 100644
index db367d8..0000000
diff --git a/swift_latest/Dockerfile b/swift_latest/Dockerfile
index 5cdeb6c..0e409f5 100644
--- a/swift_latest/Dockerfile
+++ b/swift_latest/Dockerfile
@@ -2,30 +2,28 @@ FROM ubuntu:18.04
 LABEL maintainer="Swift Infrastructure <swift-infrastructure@swift.org>"
 LABEL Description="Docker Container for the Swift programming language"
 
-# Install related packages and set LLVM 3.9 as the compiler
 RUN export DEBIAN_FRONTEND=noninteractive DEBCONF_NONINTERACTIVE_SEEN=true && apt-get -q update && \
     apt-get -q install -y \
-    make \
+    libatomic1 \
+    libbsd0 \
+    libcurl4 \
+    libxml2 \
+    libedit2 \
+    libsqlite3-0 \
     libc6-dev \
-    clang-3.9 \
-    curl \
-    libedit-dev \
+    binutils \
+    libgcc-5-dev \
+    libstdc++-5-dev \
     libpython2.7 \
-    libicu-dev \
-    libssl-dev \
-    libxml2 \
     tzdata \
     git \
-    libcurl4-openssl-dev \
     pkg-config \
-    && update-alternatives --quiet --install /usr/bin/clang clang /usr/bin/clang-3.9 100 \
-    && update-alternatives --quiet --install /usr/bin/clang++ clang++ /usr/bin/clang++-3.9 100 \
     && rm -r /var/lib/apt/lists/*
 
 # Everything up to here should cache nicely between Swift versions, assuming dev dependencies change little
 ARG SWIFT_PLATFORM=ubuntu18.04
-ARG SWIFT_BRANCH=swift-5.0-release
-ARG SWIFT_VERSION=swift-5.0-RELEASE
+ARG SWIFT_BRANCH=swift-5.0.1-release
+ARG SWIFT_VERSION=swift-5.0.1-RELEASE
 
 ENV SWIFT_PLATFORM=$SWIFT_PLATFORM \
     SWIFT_BRANCH=$SWIFT_BRANCH \
@@ -33,8 +31,12 @@ ENV SWIFT_PLATFORM=$SWIFT_PLATFORM \
 
 # Download GPG keys, signature and Swift package, then unpack, cleanup and execute permissions for foundation libs
 RUN SWIFT_URL=https://swift.org/builds/$SWIFT_BRANCH/$(echo "$SWIFT_PLATFORM" | tr -d .)/$SWIFT_VERSION/$SWIFT_VERSION-$SWIFT_PLATFORM.tar.gz \
+    && apt-get update \
+    && apt-get install -y curl \
     && curl -fSsL $SWIFT_URL -o swift.tar.gz \
     && curl -fSsL $SWIFT_URL.sig -o swift.tar.gz.sig \
+    && apt-get purge -y curl \
+    && apt-get -y autoremove \
     && export GNUPGHOME="$(mktemp -d)" \
     && set -e; \
         for key in \
diff --git a/swift_xenial/Dockerfile b/swift_xenial/Dockerfile
index c64d47c..f9acd07 100644
--- a/swift_xenial/Dockerfile
+++ b/swift_xenial/Dockerfile
@@ -2,30 +2,28 @@ FROM ubuntu:16.04
 LABEL maintainer="Swift Infrastructure <swift-infrastructure@swift.org>"
 LABEL Description="Docker Container for the Swift programming language"
 
-# Install related packages and set LLVM 3.8 as the compiler
 RUN apt-get -q update && \
     apt-get -q install -y \
-    make \
+    libatomic1 \
+    libbsd0 \
+    libcurl3 \
+    libxml2 \
+    libedit2 \
+    libsqlite3-0 \
     libc6-dev \
-    clang-3.8 \
-    curl \
-    libedit-dev \
+    binutils \
+    libgcc-5-dev \
+    libstdc++-5-dev \
     libpython2.7 \
-    libicu-dev \
-    libssl-dev \
-    libxml2 \
     tzdata \
     git \
-    libcurl4-openssl-dev \
     pkg-config \
-    && update-alternatives --quiet --install /usr/bin/clang clang /usr/bin/clang-3.8 100 \
-    && update-alternatives --quiet --install /usr/bin/clang++ clang++ /usr/bin/clang++-3.8 100 \
     && rm -r /var/lib/apt/lists/*    
 
 # Everything up to here should cache nicely between Swift versions, assuming dev dependencies change little
 ARG SWIFT_PLATFORM=ubuntu16.04
-ARG SWIFT_BRANCH=swift-5.0-release
-ARG SWIFT_VERSION=swift-5.0-RELEASE
+ARG SWIFT_BRANCH=swift-5.0.1-release
+ARG SWIFT_VERSION=swift-5.0.1-RELEASE
 
 ENV SWIFT_PLATFORM=$SWIFT_PLATFORM \
     SWIFT_BRANCH=$SWIFT_BRANCH \
@@ -33,8 +31,12 @@ ENV SWIFT_PLATFORM=$SWIFT_PLATFORM \
 
 # Download GPG keys, signature and Swift package, then unpack, cleanup and execute permissions for foundation libs
 RUN SWIFT_URL=https://swift.org/builds/$SWIFT_BRANCH/$(echo "$SWIFT_PLATFORM" | tr -d .)/$SWIFT_VERSION/$SWIFT_VERSION-$SWIFT_PLATFORM.tar.gz \
+    && apt-get -q update \
+    && apt-get -q install -y curl \
     && curl -fSsL $SWIFT_URL -o swift.tar.gz \
     && curl -fSsL $SWIFT_URL.sig -o swift.tar.gz.sig \
+    && apt-get purge -y curl \
+    && apt-get -y autoremove \
     && export GNUPGHOME="$(mktemp -d)" \
     && set -e; \
         for key in \
````

</details>

**Comment by tianon** [comment]
LGTM

Build test of #5787; 040dc596; `amd64` (`swift`):

```console

$ bashbrew build swift:5.0.1
Building bashbrew/cache:d6738cb10e4899c470905e0f500e5761d630d7b544b24482430eac04c591f9ec (swift:5.0.1)
Tagging swift:5.0.1
Tagging swift:5.0
Tagging swift:5.0.1-bionic
Tagging swift:bionic
Tagging swift:latest

$ test/run.sh swift:5.0.1
testing swift:5.0.1
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'swift-hello-world' [5/5]...passed


$ bashbrew build swift:5.0.1-xenial
Building bashbrew/cache:8da9d553444eeeeba4f75c080f92d7b34a73b1ccde2cc90dc4d8b120b1d4f3b2 (swift:5.0.1-xenial)
Tagging swift:5.0.1-xenial
Tagging swift:xenial

$ test/run.sh swift:5.0.1-xenial
testing swift:5.0.1-xenial
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'swift-hello-world' [5/5]...passed


$ bashbrew build swift:4.2.4
Using bashbrew/cache:e503303aca946405fe1d56bff03ad220ecbdb1e6590347f28afc18c4528f7c86 (swift:4.2.4)
Tagging swift:4.2.4
Tagging swift:4.2
Tagging swift:4

$ test/run.sh swift:4.2.4
testing swift:4.2.4
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'swift-hello-world' [5/5]...passed


$ bashbrew build swift:4.1.3
Using bashbrew/cache:42a12efded31efdb162d9e5ee8d011687abd3a65cb3db20aeaf84e21eaead0ef (swift:4.1.3)
Tagging swift:4.1.3
Tagging swift:4.1

$ test/run.sh swift:4.1.3
testing swift:4.1.3
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'swift-hello-world' [5/5]...passed


$ bashbrew build swift:4.0.3
Using bashbrew/cache:31f8bd997e63a2e3fbedf7f781c68e66324a582a841d1c244fa12985569269e8 (swift:4.0.3)
Tagging swift:4.0.3
Tagging swift:4.0

$ test/run.sh swift:4.0.3
testing swift:4.0.3
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'swift-hello-world' [5/5]...passed


$ bashbrew build swift:3.1.1
Using bashbrew/cache:2c16a0a73c7b7bb81055767e6108965c2bfea29779b68370be736caeb3edec14 (swift:3.1.1)
Tagging swift:3.1.1
Tagging swift:3.1
Tagging swift:3

$ test/run.sh swift:3.1.1
testing swift:3.1.1
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'swift-hello-world' [5/5]...passed

```

**Pull request merged**